packages/cocoindex/hooks/proxy_manager.py

```
@@ -58,136 +72,404 @@ def resolve_pid_path(config: dict, project_dir: str) -> str:
+        helper_path = _resolve_background_helper_path()
+        env = os.environ.copy()
+        env.setdefault("AI_ORCHESTRA_DIR", os.environ.get("AI_ORCHESTRA_DIR", ""))
+        should_spawn = True
+    finally:
+        _release_lock(lock_path)
+
+    if not should_spawn:
+        return False
+
+    try:
+        subprocess.Popen(
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)